src/alarm_scheduler.cpp

```diff
@@ -298,7 +299,8 @@ void AlarmScheduler::handle_failed_alarm(AlarmTableDef& alarm_table_def)
   TRC_DEBUG("Handling an alarm (%u) the NMS didn't respond to",
             alarm_table_def.alarm_index());
 
-  pthread_mutex_lock(&_lock);
+  // No need to take a lock - this is only called from Net-SNMP, which must
+  // already have been holding it.
```

Comment:
Should we use a recursive lock, which would negate this? If not, we should also add a function comment that this must only be called when you are already holding the lock, and note that NetSNMP satisfies this condition.

Reply:
Happy to add a function comment.  I'm a bit nervous about recursive locks in the SNMPAgent, as it is absolutely critical for performance in its other uses (e.g. in Sprout).